### PR TITLE
#167399131 User should be able to read/access articles

### PR DESCRIPTION
**Pull request by frankopkusianwar**
#### What does this PR do?

- Enable a user to read/access all articles

#### Description of Task to be completed?

- make a call to the backend API to get all the article information
- Add all the information onto the landing page

#### How should this be manually tested?

- visit ` https://ah-frontend-space-stagin-pr-19.herokuapp.com/  `to view the signup page and test the app

#### What are the relevant pivotal tracker stories?

[#167399131](https://www.pivotaltracker.com/story/show/167399131)

#### Screenshots (if appropriate)

pr articles

### CODE REVIEW

**Review by richien — COMMENTED** [review]
HI @frankopkusianwar, first of all, this is a really good effort, the articles render as mentioned in your PR. I noticed that when I click the `Recent Articles` link, there are large words the read `Loading No Articles`. Have you considered using a loader instead?
Also, I noticed the ReadMore links don't seem to work. Is this intentional?
image

**Comment by frankopkusianwar** [comment]
> HI @frankopkusianwar, first of all, this is a really good effort, the articles render as mentioned in your PR. I noticed that when I click the `Recent Articles` link, there are large words the read `Loading No Articles`. Have you considered using a loader instead?
> Also, I noticed the ReadMore links don't seem to work. Is this intentional?
> image

thanks for the review @richien as of now the words were just acting as placeholders for the loader so am working on adding the loader to resolve this. as of the readmore.. I need for first get clarity whether the scope of the story covers this as well as this involves returning one article which is a different endpoint

**Review by sebbss — CHANGES_REQUESTED** [review]
@frankopkusianwar thank you for the implementation it works fine when retrieving both getting all articles and a single article. though I notice the `like icon and unlike icon` are both pointing upwards. I suggest you consider getting one of them to point downwards as a representation of the dislikes
image

**Comment by frankopkusianwar** [comment]
> @frankopkusianwar thank you for the implementation it works fine when retrieving both getting all articles and a single article. though I notice the `like icon and unlike icon` are both pointing upwards. I suggest you consider getting one of them to point downwards as a representation of the dislikes
> image

thanks for the review @sebbss, this has been resolved

**Review by sebbss — APPROVED** [review]

**Comment by richien** [comment]
This is good @frankopkusianwar , thank you for implementing the feedback. 
Screenshot 2019-07-24 at 02 53 39
The `readmore` link now works fine but I've noticed that it's color does not match the theme of the app. Kindly rectify this.
Also, personally I think the `deepsky blue border` around the article in the article detail page does not look good. I feel the page would look a lot neater without it, what do you think? Is it also possible to have the `readtime` in a location that is more informative, for instance in the article header. It just seems abit odd to have the readtime at the end of the article.

**Comment by frankopkusianwar** [comment]
> This is good @frankopkusianwar , thank you for implementing the feedback.
> Screenshot 2019-07-24 at 02 53 39
> The `readmore` link now works fine but I've noticed that it's color does not match the theme of the app. Kindly rectify this.
> Also, personally I think the `deepsky blue border` around the article in the article detail page does not look good. I feel the page would look a lot neater without it, what do you think? Is it also possible to have the `readtime` in a location that is more informative, for instance in the article header. It just seems abit odd to have the readtime at the end of the article.

thanks for the feedback @richien, this has been worked on

**Comment by richien** [comment]
> > This is good @frankopkusianwar , thank you for implementing the feedback.
> > Screenshot 2019-07-24 at 02 53 39
> > The `readmore` link now works fine but I've noticed that it's color does not match the theme of the app. Kindly rectify this.
> > Also, personally I think the `deepsky blue border` around the article in the article detail page does not look good. I feel the page would look a lot neater without it, what do you think? Is it also possible to have the `readtime` in a location that is more informative, for instance in the article header. It just seems abit odd to have the readtime at the end of the article.
> 
> thanks for the feedback @richien, this has been worked on

Thank you @frankopkusianwar for implementing the feedback. I think that page looks a lot better, I also noticed you made a provision for images, great work on that.

**Review by richien — APPROVED** [review]

**Review by MuhanguziDavid — CHANGES_REQUESTED** [review]
Nice work with the articles @frankopkusianwar . However, I get a loader which continues loading endlessly after I have logged in. Could there be an issue with the `/articles` route?

Screenshot 2019-07-24 at 17 02 18

File: public/index.html [hunk 1]
```diff
@@ -8,7 +8,11 @@
     <title>Authors Haven</title>
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1">
-    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
+    <!-- Font Awesome -->
+<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.2/css/all.css">
+<!-- Bootstrap core CSS -->
+<link href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.3.1/css/bootstrap.min.css" rel="stylesheet">
```
Comment (MuhanguziDavid) [inline public/index.html]: Please indent these lines of code (12, 13,14).

File: src/__test__/components/HomeDetails.test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  getArticleActions: jest.fn(),
+  match: { params: { slug: 'test' } },
+};
+
+describe('App', () => {
+  it('should match snapshot', () => {
+    const wrapper = shallow(<ArticleDetails {...props} />);
+    expect(wrapper).toMatchSnapshot();
+  });
+  it('should show loading when retriving an articles', () => {
```
Comment (MuhanguziDavid) [inline src/__test__/components/HomeDetails.test.js]: please, change the word `articles` to singular.

File: public/index.html (hunk 1 above)
Comment (frankopkusianwar) [reply public/index.html]: thanks for the review @MuhanguziDavid, these changes have been worked on

File: src/__test__/components/HomeDetails.test.js (hunk 2 above)
Comment (frankopkusianwar) [reply src/__test__/components/HomeDetails.test.js]: this has been worked on @MuhanguziDavid

**Comment by frankopkusianwar** [comment]
> Nice work with the articles @frankopkusianwar . However, I get a loader which continues loading endlessly after I have logged in. Could there be an issue with the `/articles` route?
> 
> Screenshot 2019-07-24 at 17 02 18

thanks for the review @MuhanguziDavid, this came about due to routing issues however I have worked on this and it's all good

**Review by MuhanguziDavid — APPROVED** [review]
LGTM

**Pull request merged**